tls/s2n_handshake.h

```diff
@@ -183,8 +184,17 @@ struct s2n_handshake {
     /* Which handshake message number are we processing */
     int message_number;
 
+    /* State of the async pkey operation during handshake */
+    s2n_async_state async_state;
+
+    /* Indicates the CLIENT_HELLO message has been completely received */
+    unsigned client_hello_received:1;
+
+    /* Indicates the handshake blocked while trying to read or write data, and has been paused */
+    unsigned paused:1;
+
     /* Set to 1 if the RSA verification failed */
-    uint8_t rsa_failed;
+    unsigned rsa_failed:1;
```

Comment:
nit: iirc some compilers don't like a naked `unsigned`.  might want to be explicit with `unsigned int`